src/controllers/task.js

```diff
@@ -1,20 +1,36 @@
 const express = require('express');
-const TaskController = (taskModel) => {
+const TaskController = (taskModel, authService) => {
     const router = express.Router();
 
     router.get('/:id', async (req, res) => {
         const params = req.params;
         const id = parseInt(params.id, 10);
-        const [task, err] = await taskModel.getTask(id);
-        if (err) {
+
+        // Get the user_id of the user sending the request
+        const [user_id_from_request, err1] = await authService.getLoggedInUserID(req.headers);
+        if (err1) {
             return res.status(400).json({
-                data: {}, 
-                error: err.message
+                message: err1.message,
             });
         }
+
+        const [task, err2] = await taskModel.getTask(id);
+
+        if (user_id_from_request != task.user_id) {
+            // Make sure the requestor has access to this object, if not, Access Denied
+            return res.status(403).json({
+                message: "Access Denied"
+            });
+        } else if (err2) {
+            return res.status(400).json({
+                data: {},
+                error: err1.message
+            });
+        }  
+
         return res.status(200).json({
             data: task,
-            error: err ? err.message : ""
+            error: err1 ? err1.message : ""
         });
```

Comment:
Is there any way we could clean up all the different return/error handling points? Maybe a try catch?

Maybe: Try (check user_id match and if it doesn't, return access denied, and if it does, return data:task, error:err2.message)

Catch (return data empty and error:err2.message)

Let me know if that's clear or not